cruise-control-metrics-reporter/src/test/java/com/linkedin/kafka/cruisecontrol/metricsreporter/CruiseControlMetricsReporterAutoCreateTopicTest.java

```diff
@@ -107,7 +116,40 @@ public void testAutoCreateMetricsTopic() throws ExecutionException, InterruptedE
         Properties props = new Properties();
         props.setProperty(ConsumerConfig.BOOTSTRAP_SERVERS_CONFIG, bootstrapServers());
         AdminClient adminClient = AdminClient.create(props);
-        TopicDescription topicDescription = adminClient.describeTopics(Collections.singleton(TOPIC)).values().get(TOPIC).get();
+
+        // Starting with Kafka 4.0.0, the deprecated method "values()" class is completely removed from "org.apache.kafka.clients.admin.DescribeTopicsResult"
+        // so we have to use the new method "topicNameValues".
+        // To make sure that the internal build pass, this logic is introduced to choose the API based on the Kafka version
+        Method topicDescriptionMethod = null;
```

Comment:
I would do the same for the tests, we can separate the DescribeTopicsResult method detection logic into its own method, `topicNameValuesMethod()` in the `CruiseControlMetricRepoter` class, then call `topicNameValuesMethod()` in our tests to save us from the extra lines.

```suggestion
    Method topicDescriptionMethod = topicNameValuesMethod();
```